webapp/client/cypress/integration/registration.spec.js

```diff
@@ -0,0 +1,55 @@
+describe("Registration", () => {
+  beforeEach(() => {
+    cy.visit("/unlock_code_request/step/data_input");
+  });
+
+  it("submitting an empty form", () => {
+    // Submit empty form
+    cy.get("button[type=submit]").contains("Registrieren").click();
+
+    // Should have errors in the right places.
+    cy.get("[role=alert][for=dob]").contains("Dieses Feld wird benötigt");
+    cy.get("[role=alert][for=idnr]").contains("Dieses Feld wird benötigt");
+    cy.get("[role=alert][for=registration_confirm_data_privacy]").contains(
+      "Sie müssen dieses Feld auswählen, um weiter zu machen"
+    );
+    cy.get("[role=alert][for=registration_confirm_terms_of_service]").contains(
+      "Sie müssen dieses Feld auswählen, um weiter zu machen"
+    );
+    cy.get("[role=alert][for=registration_confirm_incomes]").contains(
+      "Sie müssen dieses Feld auswählen, um weiter zu machen"
+    );
+    cy.get("[role=alert][for=registration_confirm_e_data]").contains(
+      "Sie müssen dieses Feld auswählen, um weiter zu machen"
+    );
+  });
+
+  it("submitting a complete and correct form", () => {
+    cy.fixture("user").then("user", (user) => {
+      // Fill DOB
+      cy.get("input[id=dob_1]").type("3");
+      cy.get("input[id=dob_2]").type("11");
+      cy.get("input[id=dob_3]").type("1978");
+
+      // Fill tax ID
+      cy.get("input[id=idnr_1]").type("04");
+      cy.get("input[id=idnr_2]").type("531");
+      cy.get("input[id=idnr_3]").type("672");
+      cy.get("input[id=idnr_4]").type("808");
+
+      // Check boxes
+      cy.get("label[for=registration_confirm_data_privacy].checkmark").click();
+      cy.get(
+        "label[for=registration_confirm_terms_of_service].checkmark"
+      ).click();
+      cy.get("label[for=registration_confirm_incomes].checkmark").click();
+      cy.get("label[for=registration_confirm_e_data].checkmark").click();
+
+      // Submit
+      cy.get("button[type=submit]").contains("Registrieren").click();
+
+      // Should see success message
+      cy.contains("Ihre Registrierung war erfolgreich");
```

Comment:
Not disagreeing, just curious what your thoughts here are?

Reply:
I was thinking that both (text and url) might be changed by us at some point and we would need to change that test. That said, history shows that we mostly do not change urls while we change texts quite often.

Reply:
Ok, thanks for summarising that. It's what I thought but I wanted to be sure. I absolutely agree with you in this instance, but I'm still mulling over the general point of how to select elements… 

Unlike [Testing Library](https://testing-library.com/), which has a pretty refined approach to what you should query (and [appropriate support for it](https://testing-library.com/docs/queries/about)), [Cypress preferentially suggests you add `data-testid`s for selecting](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements), which I don't like too much. If you're interested in the argument, [this post matches my feelings quite well](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change), seeing testids as a last resort and preferring to interact with your application the way a user would.

Reply:
Thanks for the reading material. I agree with you that we should only use `data-testids` as a last resort. I especially liked the picture mentioned in the article to always think about what a human tester would test.
In that specific case that makes a lot of sense, too. Because a human tester would also look for the url and not for the text I would say.